File: python/findLengthOfLCIS.py
#!/usr/bin/env python3
#coding=utf-8

''' leetcode issue 674'''
class Solution:
    def findLengthOfLCIS(self, nums: List[int]) -> int:
        cur = 1
        ret = 0

        if(len(nums) == 0):
            return 0
        for i in range(1, len(nums)):
            if(nums[i] > nums[i-1]):
                cur += 1
            else:
                ret = max(ret, cur)
                cur = 1
        
        ret = max(ret, cur)
        return ret

File: c++/findLengthOfLCIS.cc
#include<vector>

using namespace std;

/* leetcode issue 674 */
class Solution {
public:
    int findLengthOfLCIS(vector<int>& nums) {
        int cur = 1;
        int ret = 0;

        if(nums.size() == 0)
            return 0;
        for(size_t i = 1; i < nums.size(); ++i) {
            if(nums[i] > nums[i-1]) {
                ++cur;
            } else {
                ret = std::max(ret, cur);
                cur = 1;
            }
        }
        ret = std::max(ret, cur);
        return ret;
    }
};
